app/frontend/js/modals/transformationDefinitionSettingsModal.js

```diff
@@ -0,0 +1,153 @@
+import { Modal } from "bootstrap";
+import editor from "../editor";
+import xmlFormat from "xml-formatter";
+import { Tooltip } from "bootstrap";
+import { each } from "lodash";
+import { bindTestForm } from "../utils/test-form";
+
+const transformationDefinitionSettingsForms = document.getElementsByClassName(
+  "js-transformation-definition-form"
+);
+
+if (transformationDefinitionSettingsForms) {
+  each(transformationDefinitionSettingsForms, (form) => {
+    initializeTransformationPreview(form);
+  });
+}
+
+function initializeTransformationPreview(form) {
```

Comment:
I think this is better, but it is still a bit hard to understand because the functions are chained, and the names don't necessarily make it clear what is happening.

I wonder if it would be clearer if you could break it down into sequential steps with good names, make functions for those, and then call all the functions sequentially.

Eg. Don't know if this is right, but from reading the code, maybe something like:

1. Find elements to update
2. Extract data
3. Update elements with data
4. Bind form (not really sure what this does)

Does that make sense?